src/focus-strategies/focus-by-distance.ts

```diff
@@ -0,0 +1,128 @@
+import { Direction } from '../model';
+
+interface IPoint {
+  x: number;
+  y: number;
+}
+
+interface IPotentialElement {
+  element: HTMLElement;
+  rect: ClientRect;
+  displacement: number;
+  score: number;
+}
+
+export function findElByDistance(direction: Direction, candidates: HTMLElement[], refRect: ClientRect): HTMLElement | null {
+  let potentialElements: IPotentialElement[] = [];
+
+  for (let i = 0; i < candidates.length; i++) {
```

Comment:
A map might be more efficient here.

Reply:
Looking at benchmark stuff, `findElByDistance` takes very little time--under 0.5ms total time in the 500 benchmark iterations. I wouldn't be too concerned here, though much of this code could be moved into one imperative loop.